crates/sui-config/src/transaction_deny_config.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+use sui_types::base_types::{ObjectID, SuiAddress};
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct TransactionDenyConfig {
+    /// A list of object IDs that are not allowed to be accessed/used in transactions.
+    /// Note that since this is checked during transaction signing, only root object ids
+    /// are supported here (i.e. no child-objects).
```

Comment:
Also no wrapped objects, and practically speaking any object whose type has `store` can also skirt around this restriction (not an issue, but worth documenting).